src/server/report/app.js

```diff
@@ -247,13 +263,33 @@ class App extends LitElement {
 				});
 				if (results.length > 0) {
 					tests.push({ ...t, results });
+					if (t.name === this._filterTest) {
+						foundFilterTest = true;
+					}
 				}
 			});
 			if (tests.length > 0) {
 				files.push({ ...f, tests });
 			}
 		});
+
+		if (this._filterTest !== undefined && !foundFilterTest) {
```

Comment:
This handles the case where you're viewing a particular test but then apply a filter that will cause that test to be excluded. We zoom back out to app home.